src/main/scala/com/imaginea/activegrid/core/models/APMServerDetails.scala

```diff
@@ -69,24 +69,22 @@ object APMServerDetails {
         try {
           val node: Node = neo4JRepository.getNodeById(nodeId)(neo)
           if (neo4JRepository.hasLabel(node, apmServerDetailsLabel)) {
-            val enityMap = collection.mutable.Map.empty[String, Option[Any]]
-            node.getRelationships.map {
-              relationship =>
-                relationship.getType.name match {
-                  case `apmServer_site_relation` =>
-                    enityMap.put("siteEntity", Site.fromNeo4jGraph(relationship.getEndNode.getId))
 
-                  case `apmServer_header_relation` =>
-                    enityMap.put("headers", Some(relationship.getEndNode.getAllProperties.foldLeft(Map[String, String]())((map, property) => map + ((property._1, property._2.asInstanceOf[String])))))
+            val tupleObj = node.getRelationships.foldLeft(Tuple2[AnyRef, Map[String, String]](AnyRef, Map.empty[String, String])) {
+              (tuple, relationsip) =>
+                val childNode = relationsip.getEndNode
+                relationsip.getType.name match {
+                  case `apmServer_site_relation` => Tuple2(Site.fromNeo4jGraph(childNode.getId), tuple._2)
```

Comment:
You don't actually have to call Tuple2 you ca just write (a,b). Same goes for type signature as well.